config/envprovider.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config // import "go.opentelemetry.io/contrib/config"
+
+import (
+	"os"
+	"regexp"
+	"strings"
+)
+
+const ValidationPattern = `^[a-zA-Z_][a-zA-Z0-9_]*$`
+
+var validationRegexp = regexp.MustCompile(ValidationPattern)
+
+func replaceEnvVar(uri string) []byte {
+	envVarName, defaultValuePtr := parseEnvVarURI(uri)
+	if !validationRegexp.MatchString(envVarName) {
+		return nil
+	}
+
+	val, exists := os.LookupEnv(envVarName)
```

Comment:
Should there be an allow list of env vars we can retrieve here, to avoid possible security problems?
For example, do we want to allow folks to inject `PATH` into the config?

Reply:
I suppose it would be up to the user if they set the configuration to include the whichever environment variable they choose to include.

It's true that someone could leak sensitive information through this mechanism, but it seems too restrictive to have an allow list. Either ways maybe this could be discussed further in the spec? Currently there are no such restrctions defined https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/data-model.md#environment-variable-substitution